torch/_inductor/pattern_matcher.py

```diff
@@ -139,6 +139,15 @@ def __init__(self) -> None:
 MULTIPLE = Multiple()
 
 
+def _transfer_meta(new_meta: Dict[str, Any], old_meta: Dict[str, Any]) -> None:
+    # transfer metadata after pattern matching occurs.
+    # skip "val" and "tensor_meta" because this info is too specific; it's unlikely
+    # to remain accurate after pattern matching has occurred.
+    new_meta.update(
+        (k, v) for k, v in old_meta.items() if k not in {"val", "tensor_meta"}
+    )
```

Comment:
Is there a pre-existing utility to do something like this? And if not - am I missing anything other than "val" and "tensor_meta" that should be skipped?

Reply:
We do have https://github.com/pytorch/pytorch/blob/9b01d17b8d7771bb0501c1389759d726b5533557/torch/fx/proxy.py#L93

although I'm not sure it's directly applicable.

The other thing is that, in some places, we do use the FakeTensorUpdater to ensure that the faketensor metadata is accurate.

Reply:
great, thanks! `_COPY_META_FIELDS` seems good enough to me. It looks like the faketensor metadata propagation is already handled in these cases and I don't see any issues with it, so I'll stick with just using `_COPY_META_FIELDS` for now.